src/app/modules/core/interceptors/mock.interceptor.ts

```diff
@@ -19,9 +22,15 @@ export class MockInterceptor implements HttpInterceptor {
 
   intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
     if (!this.environmenter.env.production) {
-      const apiIdx = request.url.indexOf('/v2/validator/');
-      if (apiIdx !== -1) {
-        const endpoint = request.url.slice(apiIdx);
+      if (this.contains(request.url, VALIDATOR_API_SUFFIX)) {
+        const endpoint = this.extractEndpoint(request.url, VALIDATOR_API_SUFFIX);
+        return of(new HttpResponse({
+          status: 200,
+          body: Mocks[endpoint],
+        }));
+      }
```

Comment:
This code repeats for the other endpoint except the constant used. I guess it will be the same for most future endpoints, and so it can be extracted into a private function.